framework/decode/dx12_dump_resources.cpp

```diff
@@ -2092,8 +2095,33 @@ void Dx12DumpResources::CopyResourceAsyncRead(graphics::dx12::ID3D12FenceComPtr
         }
     }
 
-    // After copying task is done, write the data to disk, and free it to reduce memory use.
-    active_delegate_->DumpResource(copy_resource_data);
+    auto source_resource_object_info = get_object_info_func_(copy_resource_data->source_resource_id);
+    auto source_resource             = reinterpret_cast<ID3D12Resource*>(source_resource_object_info->object);
+    auto source_resource_extra_info  = GetExtraInfo<D3D12ResourceInfo>(source_resource_object_info);
+    std::vector<graphics::dx12::ResourceStateInfo> res_infos = source_resource_extra_info->resource_state_infos;
+
+    // Create a mask of modifiable subresources.
+    // We treat D3D12_RESOURCE_STATE_COMMON as modifiable because that state could be considered
+    // modifiable based on how the resource is used. Note that D3D12_RESOURCE_STATE_COMMON is 0,
+    // so we can't just check for its bit being set
+    uint32_t modifiableResourcesBits = 0;
```

Comment:
I'm not sure if subresources count could be more than 32 or not. `uint64_t` might be better. Or it could print an error when the subresources count is more than 32.

Reply:
I can change it to 64 bits. I would think that would be enough. I don't think we should print an error. But there is probably no limit. The only other solution would be to change it to a vector. The json output for the value would be a series of integers. Thoughts?

Reply:
A vector is good. `for (const auto sub_index : resource_data->subresource_indices)` is to write every subresource data. It is just below. It could write a boolean or an integer in every subresource data.